macros/public/store/save_results_history.sql

```diff
@@ -0,0 +1,48 @@
+
+{% macro save_test_history(results) %}
+    -- depends_on: {{ ref('re_data_test_history') }}
+
+    {% set re = modules.re %}
+
+    {% if execute and results %}
+        {% set to_insert = [] %}
+        {% for el in results %}
+            {% if el.node.resource_type.name == 'Test' %}
+                {% set any_refs = re.findall("ref\(\'(?P<name>.*)\'\)", el.node.test_metadata.kwargs['model']) %}
+                {% set any_source = re.findall("source\(\'(?P<one>.*)\'\,\s+\'(?P<two>.*)\'\)", el.node.test_metadata.kwargs['model']) %}
```

Comment:
i'm wondering if `depends_on` can help here. The node contains information about its ancestors(macros/seeds/models) and it seems to me that's the thing extracted here. right?

Reply:
yeah, I was considering to use depeds_on but concern was it's more then one thing there, so we would need to make some assumption about the order they appear etc. Maybe we could take first or the last one..